src/main/java/de/hpi/bpt/argos/core/ArgosImpl.java

```diff
@@ -1,143 +1,70 @@
 	@Override
-	public void run(int port, int numberOfThreads) {
-
-		sparkService = startServer(port, numberOfThreads);
-
-		entityManager = new PersistenceEntityManagerImpl();
-		if (!entityManager.setup()) {
-			shutdown();
+	public void start() {
+		sparkService = Service.ignite()
+				.port(Argos.getPort())
+				.threadPool(Argos.getThreads())
+				.staticFileLocation(Argos.getPublicFiles());
+
+		if (!PersistenceAdapterImpl.getInstance().establishConnection()) {
+			stop();
 			return;
 		}
 
-		loadDefaultEventTypes();
-		loadBackboneData();
+		// TODO: parse default event types and entity data
+
+		// keep this order, since web sockets should be registered before any web routes get registered
+		(new ClientUpdateServiceImpl()).setup(sparkService);
+		EventProcessingPlatformUpdaterImpl.getInstance().setup();
 
-		ResponseFactory responseFactory = new ResponseFactoryImpl();
-		customerRestEndpoint = new CustomerRestEndpointImpl();
-		eventPlatformRestEndpoint = new EventPlatformRestEndpointImpl();
+		Set<RestEndpoint> restEndpoints = new HashSet<>();
+		restEndpoints.add(new EventReceiverImpl());
+		// TODO: add more restEndpoints here
 
-		// Keep this order, as spark wants to have all web sockets first
-		responseFactory.setup(entityManager, eventPlatformRestEndpoint);
-		customerRestEndpoint.setup(entityManager, sparkService, responseFactory);
-		eventPlatformRestEndpoint.setup(entityManager, sparkService, responseFactory);
+		for (RestEndpoint restEndpoint : restEndpoints) {
+			restEndpoint.setup(sparkService);
+		}
 
 		enableCORS(sparkService);
 		sparkService.awaitInitialization();
 	}
 
-    /**
-     * {@inheritDoc}
-     */
-	@Override
-	public void run() {
-		run(Argos.getPort(), Argos.getThreads());
-	}
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-	public void shutdown() {
-		sparkService.stop();
-	}
-
 	/**
 	 * {@inheritDoc}
 	 */
 	@Override
-	public void setTestMode(boolean testMode) {
-		PropertyEditor propertyEditor = new PropertyEditorImpl();
-		propertyEditor.setProperty(Argos.getArgosBackendTestModePropertyKey(), String.valueOf(testMode));
-	}
-
-	/**
-	 * {@inheritDoc}
-	 */
-	@Override
-	public PersistenceEntityManager getPersistenceEntityManager() {
-		return entityManager;
-	}
-
-	/**
-     * This method starts the Spark service on a given port with a given number of threads.
-     * @param port - port to be used as an integer
-     * @param numberOfThreads - number of threads to be used as an integer
-     * @return - returns a spark service object
-     */
-	protected Service startServer(int port, int numberOfThreads) {
-		PropertyEditor propertyEditor = new PropertyEditorImpl();
-
-		Service service = ignite()
-								.port(port)
-								.threadPool(numberOfThreads);
-
-		String publicFiles = propertyEditor.getProperty(Argos.getArgosBackendPublicFilesPropertyKey());
-		service.staticFileLocation(publicFiles);
-
-		return service;
+	public void stop() {
+		sparkService.stop();
+		System.exit(1);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this call to "exit" or ensure it is really required. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1147)